spec/support/coverage_helper.rb

```diff
@@ -0,0 +1,43 @@
+module Spec
+  module CoverageHelper
+    GEM_ROOT = Pathname.new(__dir__).join("../..").expand_path
+
+    class << self
+      attr_accessor :coverage_counter
+    end
+    self.coverage_counter = 0
```

Comment:
We should probably make a (mental) note that this is not thread safe, and if we ever find the need to parallelize this test suite, this counter will most likely cause issues

Reply:
I'll add a comment